File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnAfterCreate{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnAfterCreate{{FeatureName}}Event - AFTER CREATE event handler.
 * SRP: Handles post-creation side effects.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnAfterCreate{{FeatureName}}Event {

    private final RequestContext requestContext;

    public Object execute(Object _, Object req) {
        String userId = getUserId(req);
        log.info("AFTER CREATE {{FeatureName}} by user: {}", userId);

        // Post-creation logic:
        // - Send notifications
        // - Emit events to message queue
        // - Update related entities
        return null;
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}

File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnAfterCreate{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnAfterCreate{{FeatureName}}Event - AFTER CREATE event handler.
 * SRP: Handles post-creation side effects (notifications, logging).
 */
module.exports = {
    execute: async (_, req) => {
        try {
            const userId = req.user?.id || 'anonymous';
            const createdData = req.data;

            // Post-creation logic:
            // - Send notifications
            // - Emit events to message queue
            // - Update related entities
            // - Trigger async workflows

            cds.log('{{FeatureName}} created', {
                userId,
                data: createdData,
                timestamp: new Date().toISOString()
            });

            return createdData;
        } catch (error) {
            cds.log.error('Error in after create {{FeatureName}}: ', error);
        }
    }
};